lib/connectors/mongodb/connector.rb

```diff
@@ -93,17 +93,33 @@ def with_client
                  end
 
         begin
-          Utility::Logger.debug("Existing Databases #{client.database_names}")
-          Utility::Logger.debug('Existing Collections:')
+          databases = client.database_names
+          collections = client.collections.map(&:name)
```

Comment:
It seems like you can just use `client.collection_names`. 

https://www.mongodb.com/docs/ruby-driver/master/reference/database-tasks/

Looking at the [source code](https://github.com/mongodb/mongo-ruby-driver/blob/b70ca2df7763c7e28990c63f7907c0f9bf409cbd/lib/mongo/database/view.rb#L74) I'm assuming it's much cheaper than retrieving collections objects and iterate over them.

Reply:
I think there's a method to get collections for a particular database f.e. `client.database.collection_names`. To retrieve all collections it seems like we should use `client.collections`. But I think it doesn't make sense to check for a collection if the corresponding database doesn't exist in the first place. I'll upload another version, which makes probably more sense.